lesson4/hometask.rb

```diff
@@ -0,0 +1,23 @@
+class LazyHash < Hash
+
+  def method_missing(method_name)
+    fetch(method_name).call
+    self[method_name] = fetch(method_name).call
+  end
+
+  def lazy? method_name
+    if fetch(method_name).instance_of? Proc
```

Comment:
this condition is absolutely redundant. `instance_of?` returns `true` or `false` originally, there is no need to wrap it into additional level of condition.
so, just `fetch(method_name).instance_of? Proc`, that's it.